ReactCommon/jsi/JSCRuntime.cpp

```diff
@@ -605,11 +605,19 @@ jsi::PropNameID JSCRuntime::createPropNameIDFromString(const jsi::String& str) {
 }
 
 std::string JSCRuntime::utf8(const jsi::PropNameID& sym) {
-  return JSStringToSTLString(stringRef(sym));
+  JSStringRef strRef = stringRef(sym);
+  std::string result = JSStringToSTLString(strRef);
+  JSStringRelease(strRef);
```

Comment:
I'm not really sure if all these string releases are correct. since stringRef() does not create a new ref. It simple returns a pointer to the string ref.

Reply:
I agree, however without these JSStringRef releases, we see huge memory leak in our application, we are able to demonstrate similar behavior in the sample application. 
Secondly, there are places in the JSCRuntime.cpp which are performing JSStringRelease for similar kind pointers e.g. createValue method.

Reply:
The other JSStringRelease() usages in this files matches a String creation like this:
```cpp
jsi::PropNameID JSCRuntime::createPropNameIDFromAscii(
    const char* str,
    size_t length) {
  std::string tmp(str, length);
  JSStringRef strRef = JSStringCreateWithUTF8CString(tmp.c_str()); // Here a new string ref is created
  auto res = createPropNameID(strRef);
  JSStringRelease(strRef); // then released.
  return res;

```

Reply:
I think that the problem lays in another somewhere else and we're "fixing" it in the wrong place.

Reply:
You are right, I am not fully aware of the flow, so we made this fix based on the analysis of memory profiling, putting these fixes solved memory leaks we were seeing in our application.

Reply:
Attaching  the ios memory profile trace generated from instruments
[ios-memory-profile.trace.zip](https://github.com/facebook/react-native/files/3459127/ios-memory-profile.trace.zip)